src/test/java/com/google/alpollo/AnalysisTest.java

```diff
@@ -34,14 +42,49 @@
   private static final String LYRICS_SHORT = "I'm the mountain\n"
       + "Rising high\n" + "It's the way that I survived\n" + "I'm the mountain\n" + "Tell my tale\n"
       + "The greatest story's now for sale\n";
+  
+  private ServletContext mockServletContext = mock(ServletContext.class);
+  private HttpServletRequest request = mock(HttpServletRequest.class);
+  private HttpServletResponse response = mock(HttpServletResponse.class);
+  private SentimentServlet sentimentServletUnderTest;
+  private EntityServlet entityServletUnderTest;
+  private StringWriter responseWriter;
+
   private static final String LYRICS_WITH_METADATA = "Google has found me some nice results!";
   private static final String EMPTY_STRING = "";
   private static final String WIKI_LINK_GOOGLE = "https://en.wikipedia.org/wiki/Google";
 
 
+  @Before
+  public void setUp() throws Exception {
+    sentimentServletUnderTest = new SentimentServlet() {
+      @Override
+      public ServletContext getServletContext() {
+        return mockServletContext;
+      }
+    };
+
+    entityServletUnderTest = new EntityServlet() {
+      @Override
+      public ServletContext getServletContext() {
+        return mockServletContext;
+      }
+    };
+
+    when(mockServletContext.getResourceAsStream(anyString()))
+      .thenReturn(new FileInputStream(CONFIG_FILE_PATH));
```

Comment:
Here I was thinking of returning something like
`new FileInputStream(TEST_RESOURCE_PATH + invocation.getArgument(0))`
with `TEST_RESOURCE_PATH = System.getProperty("user.dir") + "/src/main/webapp"`

That way, you would have the possibility of accessing more than just config.json if the need arises

Reply:
aaaaa I got it, makes sense